ion-schema/src/constraint.rs

```diff
@@ -1504,6 +1545,71 @@ impl ConstraintValidator for ElementConstraint {
     }
 }
 
+/// Implements the `annotations` constraint
+/// [annotations]: https://amazon-ion.github.io/ion-schema/docs/isl-1-0/spec#annotations
+// This is used for both simple and standard syntax of annotations constraint.
+// The simple syntax will be converted to a standard syntax for removing complexity in the validation logic.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct AnnotationsConstraint2_0 {
+    type_ref: TypeReference,
+}
+
+impl AnnotationsConstraint2_0 {
+    pub fn new(type_ref: TypeReference) -> Self {
+        Self { type_ref }
+    }
+}
+
+impl ConstraintValidator for AnnotationsConstraint2_0 {
+    fn validate(
+        &self,
+        value: &IonSchemaElement,
+        type_store: &TypeStore,
+        ion_path: &mut IonPath,
+    ) -> ValidationResult {
+        let mut violations: Vec<Violation> = vec![];
+
+        match value {
+            IonSchemaElement::SingleElement(element) => {
+                let mut sequence_builder = Element::sequence_builder();
+                for annotation in element.annotations().iter() {
+                    sequence_builder =
+                        sequence_builder.push(Element::symbol(annotation.to_owned()));
+                }
+
+                let schema_element: IonSchemaElement =
+                    (&Element::from(sequence_builder.build_list())).into();
```

Comment:
I think it might work to do this.
```suggestion
                let schema_element: IonSchemaElement = 
                    element.annotations().map(Ion::symbol).collect::<Vec<_>>().into();
```